### PR TITLE
Added configurable default value to Configure::read()

### PR DESCRIPTION
**Pull request by h-moriya**
default value to `Configure::read()`

`Configure::read('key','none');` = `Configure::read('key') ?? 'none';` 

It works the same as `env ()`

### CODE REVIEW

**Comment by dereuromark** [comment]
Sounds reasonable given the other reader methods also have this default argument.

**Comment by cleptric** [comment]
What use case do you see for this feature? Do you end up missing a config option on a given case?

**Comment by dereuromark** [comment]
For BC reasons this could however go into 3.5 better
As people could have extended the Configure class and call e.g. MyConfigure::read() which would then throw notices due to the signature change.
Very theoretical, though.

**Comment by markstory** [comment]
Moved to `3.next`

**Comment by h-moriya** [comment]
@cleptric 
I think that use it in this case

https://github.com/CakeDC/ratings/blob/31136942e4dab7563b64ad3c18b96db13aaaabae/Model/Rating.php#L41
https://github.com/multidots/cakephp-rest-api/blob/867bcc5687310cf3424a03de8c11618b0cc45b15/src/Controller/AppController.php#L68
https://github.com/uzyn/cakephp-opauth/blob/2ca4dc8f887a5ea09b2536b7d1630f4db7f123b9/Controller/OpauthAppController.php#L110

File: VERSION.txt [hunk 1]
```diff
@@ -16,4 +16,4 @@
 // @license       https://opensource.org/licenses/mit-license.php MIT License
 // +--------------------------------------------------------------------------------------------+ //
 ////////////////////////////////////////////////////////////////////////////////////////////////////
-3.4.9
+3.4.10
```
Comment (ravage84) [inline VERSION.txt]: You don't need to do that. This is taken careof when it get's released.

**Review by ravage84 — COMMENTED** [review]
Lovely, that was on my TODO list for so long. Thanks for taking care of.

File: VERSION.txt (hunk 1 above)
Comment (dereuromark) [inline VERSION.txt]: this is because of branch change. can be disregarded.

**Comment by markstory** [comment]
Thanks @h-moriya :clap:

**Pull request merged**